src/Lib.P2P/Routing/DistributedQuery.cs

```diff
@@ -0,0 +1,273 @@
+                        Serializer.SerializeWithLengthPrefix(stream, queryMessage, PrefixStyle.Base128);
+                        await stream.FlushAsync(cts.Token).ConfigureAwait(false);
+                        var response = await ProtoBufHelper.ReadMessageAsync<DhtMessage>(stream, cts.Token)
+                           .ConfigureAwait(false);
+
+                        // Process answer
+                        ProcessProviders(response.ProviderPeers);
+                        ProcessCloserPeers(response.CloserPeers);
+                    }
+
+                    var time = DateTime.Now - start;
+                    log.Debug($"Q{Id}.{taskId}.{pass} ok {peer} ({time.TotalMilliseconds} ms)");
+                }
+                catch (Exception e)
+                {
+                    Interlocked.Increment(ref failedConnects);
+                    var time = DateTime.Now - start;
+                    log.Warn($"Q{Id}.{taskId}.{pass} failed ({time.TotalMilliseconds} ms) - {e.Message}");
+
+                    // eat it
+                }
+                finally
+                {
+                    askCount.Release();
+                }
+            }
+        }
+
+        private void ProcessProviders(DhtPeerMessage[] providers)
+        {
+            if (providers == null)
+                return;
+
+            foreach (var provider in providers)
+                if (provider.TryToPeer(out var p))
+                {
+                    if (p == Dht.SwarmService.LocalPeer || !Dht.SwarmService.IsAllowed(p))
+                        continue;
+
+                    p = Dht.SwarmService.RegisterPeer(p);
+                    if (QueryType == MessageType.GetProviders)
+                    {
+                        // Only unique answers
+                        var answer = p as T;
+                        if (!answers.ContainsKey(answer)) AddAnswer(answer);
+                    }
+                }
+        }
+
+        private void ProcessCloserPeers(DhtPeerMessage[] closerPeers)
+        {
+            if (closerPeers == null)
+                return;
+            foreach (var closer in closerPeers)
+                if (closer.TryToPeer(out var p))
+                {
+                    if (p == Dht.SwarmService.LocalPeer || !Dht.SwarmService.IsAllowed(p))
+                        continue;
+
+                    p = Dht.SwarmService.RegisterPeer(p);
+                    if (QueryType == MessageType.FindNode && QueryKey == p.Id) AddAnswer(p as T);
+                }
+        }
+
+        /// <summary>
+        ///   Add a answer to the query.
+        /// </summary>
+        /// <param name="answer">
+        ///   An answer.
+        /// </param>
+        /// <remarks>
+        /// </remarks>
+        public void AddAnswer(T answer)
+        {
+            if (answer == null)
+                return;
+            if (runningQuery != null && runningQuery.IsCancellationRequested)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)